bug(tool/internal/instrument): stripBuildIgnoreTag corrupts //go:build ignore text appearing anywhere in a file rule's source

### Describe the bug

`stripBuildIgnoreTag` in `tool/internal/instrument/apply_file.go` strips the `//go:build ignore` constraint from a file rule's source before it is injected into the target package, using a whole-file substring replace:

```go
func stripBuildIgnoreTag(content string) string {
	return strings.ReplaceAll(content, "//go:build ignore", "")
}
```

`strings.ReplaceAll` has no notion of Go syntax, so it deletes that text from *everywhere* in the file — string literals, doc comments, any position — not just the leading build-constraint line it is meant to remove.

The corruption is silent: the result is still valid Go, so nothing fails to compile and nothing is logged. The injected file just carries different values than its source.

### Environment

 - OS: Linux (WSL2 / Ubuntu); not platform-specific — the function is pure string manipulation
 - Go Version: 1.25.7
 - Version: `1110621` (`main`)

### To Reproduce

Steps to reproduce the behavior:

1. Add the following test to `tool/internal/instrument/`:

```go
func TestStripBuildIgnoreTagPreservesNonConstraintText(t *testing.T) {
	src := "//go:build ignore\n" +
		"\n" +
		"package hooks\n" +
		"\n" +
		"// Every file rule source must carry //go:build ignore at the top.\n" +
		"const marker = \"//go:build ignore\"\n"

	got := stripBuildIgnoreTag(src)

	if !strings.Contains(got, "const marker = \"//go:build ignore\"") {
		t.Errorf("string literal was corrupted:\n%s", got)
	}
	if !strings.Contains(got, "must carry //go:build ignore at the top") {
		t.Errorf("comment prose was corrupted:\n%s", got)
	}
	if strings.HasPrefix(strings.TrimLeft(got, "\n"), "//go:build ignore") {
		t.Errorf("leading build constraint was not stripped:\n%s", got)
	}
}
```

2. Run `go test -run TestStripBuildIgnoreTagPreservesNonConstraintText ./tool/internal/instrument/`
3. See the failure:

```text
=== RUN   TestStripBuildIgnoreTagPreservesNonConstraintText
    zz_repro_test.go:22: string literal was corrupted:

        package hooks

        // Every file rule source must carry  at the top.
        const marker = ""
    zz_repro_test.go:25: comment prose was corrupted:

        package hooks

        // Every file rule source must carry  at the top.
        const marker = ""
--- FAIL: TestStripBuildIgnoreTagPreservesNonConstraintText (0.00s)
```

`marker` becomes the empty string and the comment has a hole in it. The corrupted output still parses cleanly via `ast.NewAstParser().ParseSource`, so there is no error at any stage — the value is simply wrong at runtime.

### Expected behavior

Only genuine leading build-constraint lines are removed. Text inside string literals and comment prose is preserved verbatim.

### Additional context

**Current exposure.** The only in-tree file rule source today (`instrumentation/go.opentelemetry.io/otel/init/init_otelsdk.go`, referenced from `instrumentation/go.opentelemetry.io/otel/init/otelc.yaml`) does not contain the text outside its leading constraint, so nothing is broken right now. This is a latent trap for the next file rule whose source mentions the tag — plausible precisely because the convention requires every such file to carry it, making it a natural thing to document in a comment.

**Why existing tests do not catch it.** `TestStripBuildIgnoreTag` only supplies inputs where `//go:build ignore` appears as a standalone line (including a `multiple go:build ignore comments` case). Those pass under both the current implementation and a correct one, so they characterize current behavior rather than constrain it. No case places the text inside a string literal or mid-comment.

**Suggested fix.** Operate line-wise over the leading comment block preceding `package`, dropping only lines whose trimmed form is a build constraint, instead of substring-replacing over the whole file. Go's own `go/build` header parsing / `constraint.IsGoBuild` semantics are the reference: a `//go:build` line is only a constraint when it appears before the package clause.

--- a/tool/internal/instrument/apply_file.go
+++ b/tool/internal/instrument/apply_file.go
@@ -6,6 +6,7 @@ package instrument
 import (
 	"context"
 	"fmt"
+	"go/build/constraint"
 	"os"
 	"path/filepath"
 	"strings"
@@ -16,8 +17,19 @@ import (
 	"go.opentelemetry.io/otelc/tool/util"
 )
 
+// stripBuildIgnoreTag removes genuine "//go:build ignore" constraint lines
+// from content, line by line. It leaves every other occurrence of that text —
+// inside a string literal, inside comment prose, anywhere that isn't itself a
+// build-constraint line — untouched. See #1069: a whole-file substring
+// replace corrupted both of those.
 func stripBuildIgnoreTag(content string) string {
-	return strings.ReplaceAll(content, "//go:build ignore", "")
+	lines := strings.Split(content, "\n")
+	for i, line := range lines {
+		if constraint.IsGoBuild(line) {
+			lines[i] = ""
+		}
+	}
+	return strings.Join(lines, "\n")
 }
 
 // applyFileRule introduces the new file to the target package at compile time.

--- a/tool/internal/instrument/apply_file_test.go
+++ b/tool/internal/instrument/apply_file_test.go
@@ -66,6 +66,27 @@ func main() {}
 			input:    "",
 			expected: "",
 		},
+		{
+			// Regression test for #1069: a whole-file substring replace
+			// deleted this text from the string literal and the comment,
+			// since both happen to contain "//go:build ignore" without being
+			// a build-constraint line themselves.
+			name: "preserves the tag text inside a string literal and comment prose",
+			input: `//go:build ignore
+
+package hooks
+
+// Every file rule source must carry //go:build ignore at the top.
+const marker = "//go:build ignore"
+`,
+			expected: `
+
+package hooks
+
+// Every file rule source must carry //go:build ignore at the top.
+const marker = "//go:build ignore"
+`,
+		},
 	}
 
 	for _, tt := range tests {

